objects/src/transaction/proven_tx.rs

```diff
@@ -91,3 +93,61 @@ impl ProvenTransaction {
         self.block_ref
     }
 }
+
+// SERIALIZATION
+// ================================================================================================
+
+impl Serializable for ProvenTransaction {
+    fn write_into<W: ByteWriter>(&self, target: &mut W) {
+        self.account_id.write_into(target);
+        self.initial_account_hash.write_into(target);
+        self.final_account_hash.write_into(target);
+        target.write_u64(self.consumed_notes.len() as u64);
+        self.consumed_notes.write_into(target);
+        target.write_u64(self.created_notes.len() as u64);
+        self.created_notes.write_into(target);
+        self.tx_script_root.write_into(target);
+        self.block_ref.write_into(target);
+        self.proof.write_into(target);
+    }
+}
+
+impl Deserializable for ProvenTransaction {
+    fn read_from<R: ByteReader>(source: &mut R) -> Result<Self, DeserializationError> {
+        let account_id = AccountId::read_from(source)?;
+        let initial_account_hash = Digest::read_from(source)?;
+        let final_account_hash = Digest::read_from(source)?;
+
+        let count = source.read_u64()?;
+        let mut consumed_notes = Vec::with_capacity(count as usize);
+        for _ in 0..count {
+            consumed_notes.push(ConsumedNoteInfo::read_from(source)?);
+        }
+
+        let count = source.read_u64()?;
+        let mut created_notes = Vec::with_capacity(count as usize);
+        for _ in 0..count {
+            created_notes.push(NoteEnvelope::read_from(source)?);
+        }
+
+        let contains = source.read_bool()?;
+        let tx_script_root = match contains {
+            true => Some(Digest::read_from(source)?),
+            false => None,
+        };
```

Comment:
minor nit: I'd use a more descriptive name - e.g., `contains_tx_script` instead of just `contains`.

Reply:
replaced with `Deserializable::read_from`, this relies on https://github.com/facebook/winterfell/pull/233